hbw/app/assets/javascripts/hbw/components/form/string.js.jsx.coffee

```diff
@@ -129,14 +132,36 @@ modulejs.define 'HBWFormString', ['React', 'jQuery', 'HBWCallbacksMixin', 'HBWDe
     validationRequired: ->
       !!this.props.params.regex
 
-    updateValue: ($el) ->
+    buildVisualAndHiddenValues: (extractValueRegexp, valueParts, pattern, nextVal) ->
+      value = this.substitute(valueParts, this.getSubstitutions(extractValueRegexp, nextVal))
+
+      [@stripNonAlphanumericChars(value), CustomFormatter.applyMaskForValue(pattern, value, this.templateRegexp)]
+
+    updateValue: ($el, remove) ->
       if @extractValueRegexp
-        substitutions = this.getSubstitutions(@extractValueRegexp, $el.val())
-        value = this.substitute(@valueParts, substitutions)
-        newValue = @stripNonAlphanumericChars(value)
-        @setState(value: newValue)
+        visualValue = this.state.visualValue
```

Comment:
I see `this.` in some places and `@` in other. Do we have any convention about this sugar?